consul/state/schema.go

```diff
@@ -0,0 +1,351 @@
+	return &memdb.TableSchema{
+		Name: "kvs",
+		Indexes: map[string]*memdb.IndexSchema{
+			"id": &memdb.IndexSchema{
+				Name:         "id",
+				AllowMissing: false,
+				Unique:       true,
+				Indexer: &memdb.StringFieldIndex{
+					Field:     "Key",
+					Lowercase: false,
+				},
+			},
+			"session": &memdb.IndexSchema{
+				Name:         "session",
+				AllowMissing: true,
+				Unique:       false,
+				Indexer: &memdb.StringFieldIndex{
+					Field:     "Session",
+					Lowercase: false,
+				},
+			},
+		},
+	}
+}
+
+// tombstonesTableSchema returns a new table schema used for
+// storing tombstones during KV delete operations to prevent
+// the index from sliding backwards.
+func tombstonesTableSchema() *memdb.TableSchema {
+	return &memdb.TableSchema{
+		Name: "tombstones",
+		Indexes: map[string]*memdb.IndexSchema{
+			"id": &memdb.IndexSchema{
+				Name:         "id",
+				AllowMissing: false,
+				Unique:       true,
+				Indexer: &memdb.StringFieldIndex{
+					Field:     "Key",
+					Lowercase: false,
+				},
+			},
+		},
+	}
+}
+
+// sessionsTableSchema returns a new TableSchema used for
+// storing session information.
+func sessionsTableSchema() *memdb.TableSchema {
+	return &memdb.TableSchema{
+		Name: "sessions",
+		Indexes: map[string]*memdb.IndexSchema{
+			"id": &memdb.IndexSchema{
+				Name:         "id",
+				AllowMissing: false,
+				Unique:       true,
+				Indexer: &memdb.StringFieldIndex{
+					Field:     "ID",
+					Lowercase: false,
+				},
+			},
+			"node": &memdb.IndexSchema{
+				Name:         "node",
+				AllowMissing: true,
+				Unique:       false,
+				Indexer: &memdb.StringFieldIndex{
+					Field:     "Node",
+					Lowercase: true,
+				},
+			},
+		},
+	}
+}
+
+// sessionChecksTableSchema returns a new table schema used
+// for storing session checks.
+func sessionChecksTableSchema() *memdb.TableSchema {
+	return &memdb.TableSchema{
+		Name: "session_checks",
+		Indexes: map[string]*memdb.IndexSchema{
+			"id": &memdb.IndexSchema{
+				Name:         "id",
+				AllowMissing: false,
+				Unique:       true,
+				Indexer: &memdb.CompoundIndex{
+					Indexes: []memdb.Indexer{
+						&memdb.StringFieldIndex{
+							Field:     "Node",
+							Lowercase: true,
+						},
+						&memdb.StringFieldIndex{
+							Field:     "CheckID",
+							Lowercase: true,
+						},
+						&memdb.StringFieldIndex{
+							Field:     "Session",
+							Lowercase: false,
+						},
+					},
+				},
+			},
+			"node_check": &memdb.IndexSchema{
+				Name:         "node_check",
+				AllowMissing: false,
+				Unique:       false,
+				Indexer: &memdb.CompoundIndex{
+					Indexes: []memdb.Indexer{
+						&memdb.StringFieldIndex{
+							Field:     "Node",
+							Lowercase: true,
+						},
+						&memdb.StringFieldIndex{
+							Field:     "CheckID",
+							Lowercase: true,
+						},
+					},
+				},
+			},
+			"session": &memdb.IndexSchema{
+				Name:         "session",
+				AllowMissing: false,
+				Unique:       true,
```

Comment:
I don't think this will be unique, since a single session could have multiple entries